File: projekt3/World.py

from Pineborscht import Pineborscht

class World:
    __size__ = None
    __map__ = None
    __events__ = None
    __turn__ = 1
    __human__ = None
    
    def __init__(self, size):
        self.__size__ = size
        self.__map__ = [[None for i in range(self.__size__)] for j in range(self.__size__)]
        self.__events__ = []

    def getHuman(self):
        return self.__human__
    
    def setHuman(self, human):
        self.__human__ = human
        
    def isInsideWorld(self, row, col):
        if(row <0 or row >=self.__size__ or col <0 or col >=self.__size__):
            return False
        else:
            return True
        
    def getTurn(self):
        return self.__turn__
    
    def incrementTurn(self):
        self.__turn__ += 1
        
    def size(self):
        return self.__size__
    
    def isPosEmpty(self, row, col):
        if(self.__map__[row][col] == None):
            return True      
        else:  
            return False
    
    def getOrgFromPos(self, row, col):
        return self.__map__[int(row)][int(col)]
    
    def removeOrganism(self, row, col):
        self.__map__[int(row)][int(col)] = None
        
    def getOrgs(self):
        orgs = []
        for i in range(self.__size__):
            for j in range(self.__size__):
                if(self.__map__[i][j] != None):
                        orgs.append(self.__map__[i][j])
        return orgs
    
    def getPines(self):
        pines = []
        for i in range(self.__size__):
            for j in range(self.__size__):
                if(self.__map__[i][j] != None  and isinstance(self.__map__[i][j], Pineborscht)):
                        pines.append(self.__map__[i][j])
        return pines
        
    def setOrganism(self, row, col, Organism):
        self.__map__[row][col] = Organism
        
    def getMap(self):
        return self.__map__
        
    def getEvents(self):
        return self.__events__
    
    def clearEvents(self):
        self.__events__.clear()
        
    def addEvent(self, message):
        self.__events__.append(message)
        
    def setTurn(self, turn):
        self.__turn__ = turn
        
    def saveToString(self):
        return str(self.__size__) + ' ' + str(self.__turn__) + '\n'  

File: projekt2/src/proj/java/model/World.java
package proj.java.model;

import java.util.ArrayList;
import java.util.List;

import proj.java.model.animal.Human;

public class World {
    private final int size;
    private final Organism[][] cells;
    private final List<String> messages;
    private int turn = 1;
    Human human = null;

    public World(int size) {
        this.size = size;
        this.cells = new Organism[size][size];
        for (var r = 0; r < size; r++) {
            var row = new Organism[size];
            this.cells[r] = row;
        }
        this.messages = new ArrayList<>();
    }

    public World(String line) {
        var fragments = line.split(" ");
        this.size = Integer.parseInt(fragments[0]);
        this.cells = new Organism[size][size];
        for (var r = 0; r < size; r++) {
            var row = new Organism[size];
            this.cells[r] = row;
        }
        this.messages = new ArrayList<>();
        this.turn = Integer.parseInt(fragments[1]);

    }

    public Human getHuman() {
        return human;
    }

    public void setHuman(Human human) {
        this.human = human;
    }

    public boolean isCellNextToHuman(int row, int col) {
        return col == human.getCol() && row == human.getRow() - 1
                || col == human.getCol() && row == human.getRow() + 1
                || col == human.getCol() - 1 && row == human.getRow()
                || col == human.getCol() + 1 && row == human.getRow();
    }

    public boolean isInsideWorld(int row, int col) {
        if (row < 0 || row >= size || col < 0 || col >= size)
            return false;
        else
            return true;
    }

    public int getTurn() {
        return turn;
    }

    public void incrementTurn() {
        turn++;
    }

    public int size() {
        return size;
    }

    public boolean isCellEmpty(int row, int col) {
        return getOrganism(row, col) == null;
    }

    public Organism getOrganism(int row, int col) {
        return cells[row][col];
    }

    public void removeOrganism(int row, int col) {
        cells[row][col] = null;
    }

    public void setOrganism(int row, int col, Organism org) {
        org.setRow(row);
        org.setCol(col);
        cells[row][col] = org;
    }

    public List<String> getMessages() {
        return messages;
    }

    public void clearMessges() {
        messages.clear();
    }

    public void addMessage(String message) {
        messages.add(message);
    }

    public void setTurn(int turn) {
        this.turn = turn;
    }

    public String writeString() {
        return String.format("%d %d", size, turn);
    }
}
